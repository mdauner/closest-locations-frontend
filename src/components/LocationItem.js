import React from 'react'
import PropTypes from 'prop-types'
import { getFormattedDistance } from '../utils'
import './LocationItem.scss'

function LocationItem({ location, distance, closestLocation }) {
  return (
    <div className="location" key={location} data-testid="location-item">
      <div className="location__from">{location}</div>
      <div className="location__distance">{getFormattedDistance(distance)}</div>
      <div className="location__to">{closestLocation}</div>
    </div>
  )
}

LocationItem.propTypes = {
  location: PropTypes.string.isRequired,
  distance: PropTypes.number.isRequired,
  closestLocation: PropTypes.string.isRequired
}

export default LocationItem
