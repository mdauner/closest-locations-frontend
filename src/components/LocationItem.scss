.location {
  width: 100%;
  background-color: white;
  padding: 1rem;
  margin-bottom: .3rem;
  display: flex;
  box-sizing: border-box;

  &:not(:first-of-type) {
    margin-top: .3rem;
  }

  &__from,
  &__to {
    flex: 1;
  }

  &__to {
    text-align: right;
  }
}