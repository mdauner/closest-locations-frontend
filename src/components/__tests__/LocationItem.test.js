import React from 'react'
import { render } from 'react-testing-library'
import LocationItem from '../LocationItem'

test('Renders LocationItem correctly', () => {
  const { container } = render(
    <LocationItem
      location="Berlin"
      distance={123456}
      closestLocation="Hamburg"
    />
  )
  expect(container.firstChild).toMatchSnapshot()
})
