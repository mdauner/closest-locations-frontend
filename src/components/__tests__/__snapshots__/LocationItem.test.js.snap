// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders LocationItem correctly 1`] = `
<div
  class="location"
  data-testid="location-item"
>
  <div
    class="location__from"
  >
    Berlin
  </div>
  <div
    class="location__distance"
  >
    76.71 mi
  </div>
  <div
    class="location__to"
  >
    Hamburg
  </div>
</div>
`;
